blockchain/system/kip113.go

```diff
@@ -30,15 +32,50 @@ import (
 type BlsPublicKeyInfo struct {
 	PublicKey []byte
 	Pop       []byte
+	VerifyErr error // Nil if valid. Must check before use.
 }
 
-type (
-	BlsPublicKeyInfos map[common.Address]BlsPublicKeyInfo
-	AllocKip113Init   struct {
-		Infos BlsPublicKeyInfos
-		Owner common.Address
+func newBlsPublicKeyInfo(publicKey []byte, pop []byte) BlsPublicKeyInfo {
+	return BlsPublicKeyInfo{
+		PublicKey: publicKey,
+		Pop:       pop,
+		VerifyErr: verifyBlsPublicKeyInfo(publicKey, pop),
 	}
-)
+}
+
+func verifyBlsPublicKeyInfo(publicKey []byte, pop []byte) error {
+	pk, err := bls.PublicKeyFromBytes(publicKey)
+	if err != nil {
+		return err
+	}
+
+	sig, err := bls.SignatureFromBytes(pop)
+	if err != nil {
+		return err
+	}
+
+	if !bls.PopVerify(pk, sig) {
+		return ErrKip113BadPop
```

Comment:
Can the error name be polymorphic, rather than the specific error for KIP113?

Reply:
Could be, but so far PoP verification happens only here in the whole codebase. When we find the second PopVerify use-case then let's separate like
```go
errBadPop = errors.New("PoP verification failed")
fmt.Errorf("KIP113 error: %s", errBadPop)
fmt.Errorf("Something other error: %s", errBadPop)
```
But not now.